lib/_http_server.js

```diff
@@ -482,8 +483,15 @@ function connectionListener(socket) {
 
       res.detachSocket(socket);
 
-      if (res._last) {
-        socket.destroySoon();
+      if (res._last) socket.destroySoon();
+      else if (outgoing.length === 0) {
+        // When outgoing queue is empty we will wait for
+        // server.keepAliveTimeout and destroy socket not emiting
+        // 'timeout' events in req, res and server objects
+        socket.removeAllListeners('timeout');
+        socket.setTimeout(self.keepAliveTimeout, function onKeepAliveTimeout() {
```

Comment:
@tshemsedinov isn't it still a valid reason to send 408?

```
The client did not produce a request within the time that the server was prepared to wait.
```

Closed connection is closed connection, we are handling a dangling TCP connection right now.